### DISCUSSION TITLE
Acceptor horizontal scaling

### DISCUSSION THREAD

**Discussion by kiropowered** [quickfix-j/quickfixj#759] [Q&A]
Hello,
I am wondering how you can achieve horizontal scaling of an acceptor server.
Let say I have 5 servers, each one has a session opened to it and they want to communicate.
Sending a message from server 1(session A) to server2(session B) will not work as the server 1 does not have the session B connected to it.
Let's assume as well I want to allow only one connection from a session, how do you prevent it in this scenario ?

**Comment by chrjohn** [comment]
Sorry it is not entirely clear to me what you want to achieve. You say acceptor server and want to communicate from server 1 to server 2. But in FIX protocol communication goes between acceptor and initiator.
Maybe you could draw a diagram?

**Comment by kiropowered** [comment]
We do the actual message processing in an external app(worker), in order to do that messages are sent through a queue system then processed.
When processes are done they sent back responses to the fix engine through a queue as well.

Problem is if we try to provide high availability on the fix engine it means with have multiple instances and sinces the sessions are in memory if a response is handled by a fix engine A but the connected session is actually on fix engine B then it is not found

Basically the fix engine is only used as proxy and don't do any business logic

We actually implement some kind of broadcasting with redis, like the socket io [[https://socket.io/docs/v4/redis-adapter/](url) ](https://socket.io/docs/v4/redis-adapter/) does with pub/sub mechanism, but since our storage is a mysql dbtabase i thought there would be some kind of option to provide working with multiple instances

**Reply by chrjohn** [reply]
How does your session config look like?

But a FIX session is only a point-to-point connection. One initiator can only be connected to one acceptor at a time.
Or is only one of your acceptors connected at any time? If you have a database then each acceptor can use the DB as session store (but only **one** acceptor at any time).
